### PR TITLE
feat: Use Apache HTTPClient for downloads of public resources

### PR DESCRIPTION
**Pull request by aikebah**
## Description of Change

Swap out use of RAW JDK http(s) handling by Apache HTTP-Client in order to also take into account the `https.proxyUser` and `https.proxyPassword` JAVA_TOOL_OPTIONs

Add a resources to document and support manual testing of HTTP-Basic Authenticating proxy usage.

Also lays the groundwork for resolving #5387 by adding Settings-keys for the credentials of a mirrored HostedSuppressions- and KnownExploitedVulnerabilities-file. CLI instances initialized from a custom properties file could already exploit this, but activation for all integrations is left as an excercise in the context of #5387.

For now only a draft, as there is still some direct raw usage left behind in the codebase that I also intend to replace by HTTPClient. However I think now is a good time to already collect feedback on the initial coding for the replacement of the Downloader class

## Have test cases been added to cover the new functionality?

Not yet, intend to evaluate and potentially extend the existing testcases that already partially test the functionality

### CODE REVIEW

File: utils/src/main/java/org/owasp/dependencycheck/utils/Downloader.java [hunk 1]
```diff
@@ -13,203 +13,494 @@
...
+     * @throws TooManyRequestsException                                thrown when a 429 is received
+     * @throws ResourceNotFoundException                               thrown when a 404 is received
      */
     public void fetchFile(URL url, File outputPath, boolean useProxy) throws DownloadFailedException,
             TooManyRequestsException, ResourceNotFoundException {
-        fetchFile(url, outputPath, useProxy, null, null);
+        try {
+            if ("file".equals(url.getProtocol())) {
+                final Path p = Paths.get(url.toURI());
+                Files.copy(p, outputPath.toPath(), StandardCopyOption.REPLACE_EXISTING);
```
[the remark this answers is not in the document]
Comment (jeremylong) [reply utils/src/main/java/org/owasp/dependencycheck/utils/Downloader.java:297]: Due to the nature of ODC being a CLI and this only occurs using the "file://" protocol - this is not considered a risk.

**Comment by jeremylong** [comment]
@aikebah tests are passing - should this still be a draft PR?

**Comment by aikebah** [comment]
@jeremylong All code as committed I consider final, so if you're eager to get a release out it would be fine to remove draft status and get 'most usages replaced' as feature for 11.0

The draft status is not-yet removed as there are a few analyzers that don't use the HTTPClient yet - Nexus and Artifactory IIRC, which could also be postponed for a later feature-update I guess.

**Comment by aikebah** [comment]
Confirmed... only NexusSearch and ArtifactorySearch are pending. As both are typically solutions within the enterprise datacenter not requiring proxy access I think it would be fine to finish up this part and leave the remainder phase-out of URLConnectionFactory for a later feature-release in the 11.x series.

**Review by jeremylong — APPROVED** [review]
LGTM

**Comment by jeremylong** [comment]
Thanks for the work on this PR! It looks like you have completed the groundwork for the remaining two analyzers (as seen in the `CentralAnalyzer`). Really appreciate all the work you put into the project!

**Comment by aikebah** [comment]
Was a nice exercise for part of my holiday. NexusSearch currently cooking locally on lower velocity besides my daytime job.

**Pull request merged**